crates/swc_ecma_utils/src/lib.rs

```diff
@@ -1263,6 +1267,17 @@ pub trait ExprExt {
                             sym: js_word!("Math"),
                             ..
                         }) => span.ctxt == SyntaxContext::empty(),
+
+                        // Some methods of string are pure
```

Comment:
I think we could simplify this entirely, all methods should be pure. Same for methods on the other primitives?

Reply:
Oh, you are right. NIce catch and thanks!

Reply:
Ah, I missed one point.
`String.replace(/asd/, ()=>{})` can have side effects